### PR TITLE
mypy for monitor exporter

### PR DESCRIPTION
**Pull request by lzchen**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/32613
Fixes https://github.com/Azure/azure-sdk-for-python/issues/32825

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/metrics/_exporter.py [hunk 1]
```diff
@@ -265,9 +268,4 @@ def _is_status_code_success(status_code: Optional[str], threshold: int) -> bool:
 def _get_metric_export_result(result: ExportResult) -> MetricExportResult:
     if result == ExportResult.SUCCESS:
         return MetricExportResult.SUCCESS
-    if result in (
-        ExportResult.FAILED_RETRYABLE,
-        ExportResult.FAILED_NOT_RETRYABLE,
-    ):
-        return MetricExportResult.FAILURE
-    return None
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/metrics/_exporter.py:273]: What's the reason for this change?

Comment (lzchen) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/metrics/_exporter.py:273]: Not really necessary to return None anymore because there are no other cases besides the three `ExportResult`s

**Review by jeremydvoss — APPROVED** [review]
